bokeh/models/graphs.py

```diff
@@ -55,17 +55,37 @@ def from_networkx(graph, layout_function, **kwargs):
         from ..models.graphs import StaticLayoutProvider
 
         # Handles nx 1.x vs 2.x data structure change
-        nodes = list(graph.nodes())
-        edges = list(graph.edges())
-
-        edges_start = [edge[0] for edge in edges]
-        edges_end = [edge[1] for edge in edges]
-
-        node_source = ColumnDataSource(data=dict(index=nodes))
-        edge_source = ColumnDataSource(data=dict(
-            start=edges_start,
-            end=edges_end
-        ))
+        # Convert node attributes
+        node_dict = dict()
+        node_dict['index'] = list(graph.nodes())
+
+        node_attr_keys = [attr_key for node in list(graph.nodes(data=True))
+                          for attr_key in node[1].keys()]
+        node_attr_keys = list(set(node_attr_keys))
+
+        for attr_key in node_attr_keys:
+            node_dict[attr_key] = [node_attr[attr_key]
+                                   if attr_key in node_attr.keys() else None
```

Comment:
prefer if conditional expressions don't span multiple lines (our linter line limit is 140)